Modified usages of sets.String{} to use the new generic type all across the kubernetes scheduler.(Waiting for generics to be part of the current supported releases to merge))

Bryce-Soghigian opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?
/kind cleanup


#### What this PR does / why we need it:
WE deprecated sets.String to be replaced with the [generic](https://github.com/kubernetes/kubernetes/pull/112377) implementation of Set.

This PR is for the sig-scheduling cleanup and moving over to the new generic types.
#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #113709

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

Aut0R3V commented on pkg/scheduler/internal/queue/scheduling_queue.go:
```diff
@@ -191,8 +191,8 @@ type priorityQueueOptions struct {
 	podNominator                      framework.PodNominator
-	clusterEventMap                   map[framework.ClusterEvent]sets.String
 	preEnqueuePluginMap               map[string][]framework.PreEnqueuePlugin
+	clusterEventMap                   map[framework.ClusterEvent]sets.Set[string]
```
I think there are some checks that have not been approved. Could you check them out?

Bryce-Soghigian replied:
What do you mean checks? I ran go test here and it passed.

sanposhiho reviewed (commented):
Please restore removed files under `/vendor`. I guess you removed them mistakenly.

sanposhiho commented:
/ok-to-test

kerthcet commented:
@Bryce-Soghigian Plz fix the CI first, thanks a lot.

Bryce-Soghigian commented:
/retest-required

Bryce-Soghigian commented:
/retest-required

Commit:
modified types for scheduling_queue

e2e scheduling modifications

modifying pref scheduler set instances

adding back v1

fixing vendor deps

fixing unit tests

fixing unit-tests

fixing list() calls

fixing vendor

fixing preempt test

Modifying tests to use Sets.List()

fixing ubernetes

Bryce-Soghigian commented:
@kerthcet  @sanposhiho @logicalhan I have addressed all of the CI problems. Can i get a LGTM label and or an Approval label?

Bryce-Soghigian commented:
/cc https://github.com/orgs/kubernetes/teams/sig-scheduling-leads

kerthcet commented on pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go:
```diff
@@ -1734,8 +1734,14 @@ func TestPreempt(t *testing.T) {
 			}
-			if diff := cmp.Diff(patchedPodNames.List(), deletedPodNames.List()); diff != "" {
-				t.Errorf("unexpected difference in the set of patched and deleted pods: %s", diff)
+			patchedPodNamesList := getListOfZones(patchedPodNames)
```
Use sets.List() instead.

Bryce-Soghigian replied:
modified code to use sets.List()

kerthcet commented on pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go:
```diff
@@ -1734,8 +1734,14 @@ func TestPreempt(t *testing.T) {
+			patchedPodNamesList := getListOfZones(patchedPodNames)
+			deletedPodNamesList := getListOfZones(deletedPodNames)
+
+			if !patchedPodNames.HasAll(deletedPodNamesList...) {
```
I think it's better to use cmp.Diff()

kerthcet commented on test/e2e/scheduling/ubernetes_lite.go:
```diff
@@ -132,9 +132,9 @@ func SpreadServiceOrFail(f *framework.Framework, replicaCount int, zoneNames set
+	zones := getListOfZones(zoneNames)
 	// Now make sure they're spread across zones
-	checkZoneSpreading(f.ClientSet, pods, zoneNames.List())
+	checkZoneSpreading(f.ClientSet, pods, zones)
```
ditto.

kerthcet commented on test/e2e/scheduling/ubernetes_lite.go:
```diff
@@ -236,6 +236,15 @@ func SpreadRCOrFail(f *framework.Framework, replicaCount int32, zoneNames sets.S
+}
+
+// getListOfZones converts a zones set to a list of strings.
+func getListOfZones(zones sets.Set[string]) (output []string) {
```
Remove this.

kerthcet commented on test/e2e/storage/ubernetes_lite_volumes.go:
```diff
@@ -147,3 +149,11 @@ func PodsUseStaticPVsOrFail(f *framework.Framework, podCount int, image string)
 	}
 }
+
+// getListOfZones converts a zones set to a list of strings.
```
ditto

Bryce-Soghigian commented:
/retest-required

Bryce-Soghigian commented:
/retest

Bryce-Soghigian commented:
/test pull-kubernetes-e2e-gce-storage-snapshot

Bryce-Soghigian commented:
/assign @damemi @pohly

pohly commented:
Change looks good, but needs to be rebased.

Bryce-Soghigian commented:
@kerthcet  I am looking for that LGTM label again. WE waited for generics to be supported, but it seems that all supported k8s versions right now are using go 1.18 so we can probably merge this in

Bryce-Soghigian commented:
/retest-required

kerthcet commented:
fix the tests first?

kidddddddddddddddddddddd commented:
@Bryce-Soghigian are you coming back for this?

kerthcet commented:
Kindly ping @Bryce-Soghigian , would you mind to revisit this again? Or if you do not have bandwidth, the other contributors from the community can help with this and make you a co-author, for we have several PRs want to address the same issue.

Bryce-Soghigian commented:
> Kindly ping @Bryce-Soghigian , would you mind to revisit this again? Or if you do not have bandwidth, the other contributors from the community can help with this and make you a co-author, for we have several PRs want to address the same issue.

Its ok other people can put in their prs

kerthcet commented:
kindly ping @yoongon @kidddddddddddddddddddddd @abhishekkhanna2710 who volunteers to address the issue, also plz make @Bryce-Soghigian a co-author as his great effort.
